File: _정답코드/1주차/05_삼각형_최소_둘레_구하기/solution.py
def solution(sides):
    n = len(sides)
    minP = float('inf')
    for i in range(n-2):
        for j in range(i+1, n-1):
            for k in range(j+1, n):
                a, b, c = sides[i], sides[j], sides[k]
                if a + b > c and a + c > b and b + c > a:
                    s = a + b + c
                    if s < minP:
                        minP = s
    assert minP != float('inf'), "No valid triangle can be formed"
    return minP

File: _정답코드/1주차/05_삼각형_최소_둘레_구하기/solution.cpp
#include <vector>
#include <string>
using namespace std;

int solution(vector<int> sides) {
    int n = sides.size();
    int minP = 30001;
    for(int i = 0; i < n - 2; i++){
        for(int j = i + 1; j < n - 1; j++){
            for(int k = j + 1; k < n; k++){
                int a = sides[i], b = sides[j], c = sides[k];
                if(a + b > c && a + c > b && b + c > a){
                    int sum = a + b + c;
                    if(sum < minP) minP = sum;
                }
            }
        }
    }
    return minP;
}